templates/layout.html

```diff
@@ -4,32 +4,140 @@
     <meta charset="UTF-8">
     <title>{% block pagetitle %}{% endblock %} | ICTM Teaching</title>
     <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='bootstrap.min.css') }}">
+    <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='jquery-ui.min.css') }}">
+    <link href="{{ url_for('static', filename='select2.min.css') }}" rel="stylesheet">
+    <link href="{{ url_for('static', filename='fontawesome-6.5.2/css/all.min.css') }}" rel="stylesheet">
+
+    <script src="{{ url_for('static', filename='jquery-3.7.1.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='jquery-ui.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='popper.min.js') }}"></script>
     <script src="{{ url_for('static', filename='bootstrap.min.js') }}"></script>
+    <script src="{{ url_for('static', filename='select2.min.js') }}"></script>
+
+    <style>
+        .form-select {
+            width: auto !important;
+        }
+    </style>
+    <script>
+        $(document).ready(function () {
+            $('#newYear').click(function () {
+                $('#confirmationModal').modal('show');
+            });
+
+            $('#confirmChange').click(function () {
```

Comment:
You can also put a POST form into the modal code and use a submit button, that would greatly simplify the code.